src/datahike/config.cljc

```diff
@@ -1,26 +1,89 @@
 (ns datahike.config
-  (:require [clojure.spec.alpha :as s])
+  (:require [clojure.edn :as edn]
+            [clojure.spec.alpha :as s]
+            [environ.core :refer [env]])
   (:import [java.net URI]))
 
-
-(s/def ::username string?)
-(s/def ::password string?)
-(s/def ::path string?)
-(s/def ::backend keyword?)
-(s/def ::host string?)
-(s/def ::port int?)
-(s/def ::uri string?)
+(s/def ::backend #{:mem :file :level :pg})
+(s/def ::username (s/nilable string?))
+(s/def ::password (s/nilable string?))
+(s/def ::path (s/nilable string?))
+(s/def ::host (s/nilable string?))
+(s/def ::port (s/nilable int?))
+(s/def ::uri (s/nilable string?))
 
 (s/def ::schema-on-read boolean?)
 (s/def ::temporal-index boolean?)
 
-(s/def :datahike/config (s/keys :req-un [::backend]
-                                :opt-un [::host ::port ::path ::username ::password]))
+(s/def :datahike/store (s/keys :req-un [::backend]
+                               :opt-un [::username ::password ::path ::host ::port]))
+(s/def :datahike/schema (s/keys :opt-un [::schema-on-read ::temporal-index]))
```

Comment:
namespaces will be changed in the future. So we would have something like `:index/schema-flexibility` with better naming.

Reply:
should I make schema-on-read and temporal-index first level options? Means taking schema-record out.